dali/operators/generic/reshape.cc

```diff
@@ -370,6 +391,21 @@ void Reshape<GPUBackend>::RunImpl(DeviceWorkspace &ws) {
   out.SetLayout(layout);
 }
 
+template <typename Backend>
+void Reshape<Backend>::CheckSrcDims(const Workspace &ws) {
+  if (src_dims_.empty()) {
+    return;
+  }
+
+  const auto &in = ws.template InputRef<Backend>(0);
+  const auto &input_shape = in.shape();
+  const int ndim = input_shape.sample_dim();
+  for (const auto src_dim : src_dims_) {
+    DALI_ENFORCE(-1 <= src_dim && src_dim < ndim,
+      make_string(OpName(), ": dimension in src_dims is out of range"));
```

Comment:
```suggestion
      make_string(OpName(), ": Out of bounds ``src_dims`` index. The indices in ``src_dims`` should be either a valid dimension index (range 0..ndim-1) or -1 to insert a new dimension. Got: src_dims[", d, "]=", src_dim, ", ", ndim=", ndim));
```
Something like this would give more useful info to the user